File: 2348-number-of-zero-filled-subarrays/2348-number-of-zero-filled-subarrays.java
class Solution {
    public long zeroFilledSubarray(int[] nums) {

        long ans=0, curr=0;
        for(int i=0;i<nums.length; i++){
            curr=(nums[i]==0? curr +1 : 0);
            ans+=curr;
            
        }
        return ans;
    }
}
    

File: 2348-number-of-zero-filled-subarrays/2348-number-of-zero-filled-subarrays.cpp
class Solution {
public:
    long long zeroFilledSubarray(vector<int>& nums) {
        long long int ans=0, curr=0;
        for(auto num: nums){
            curr=(num==0? curr +1 : 0);
            ans+=curr;
            
        }
        return ans;
    }
};